src/pages/VaultsPage/VaultTransactions.tsx

```diff
@@ -1,62 +1,285 @@
-import { client } from 'lib/gql/client';
-import { allVaultFields } from 'lib/gql/mutations';
+import { useState, useEffect } from 'react';
+
+import { ContractsReadonly } from 'common-lib/contracts';
+import { BigNumber } from 'ethers';
+import { getCircle, getVaultAndTransactions } from 'lib/gql/queries';
+import { decodeCircleId } from 'lib/vaults';
+import { DateTime } from 'luxon';
 import { useQuery } from 'react-query';
 import { useParams } from 'react-router-dom';
 import { styled } from 'stitches.config';
 
 import { Link, Panel, Text } from 'ui';
 import { OrgLayout, SingleColumnLayout } from 'ui/layouts';
+import { getProviderForChain } from 'utils/provider';
+
+import { Awaited } from 'types/shim';
+
+type VaultAndTransactions = Awaited<ReturnType<typeof getVaultAndTransactions>>;
 
 export const VaultTransactions = () => {
   const { address } = useParams();
-  const { isLoading, isIdle, data } = useQuery('vault', () =>
-    client.query(
-      {
-        vaults: [
-          { where: { vault_address: { _eq: address } } },
-          {
-            ...allVaultFields,
-            protocol: {
-              name: true,
-            },
-          },
-        ],
-      },
-      { operationName: 'getVault' }
-    )
-  );
-
-  const vault = data?.vaults[0];
+  const [vaultTxList, setVaultTxList] = useState<any>([]);
+  const {
+    isLoading,
+    isIdle,
+    data: vault,
+  } = useQuery(['vault', address], async () => {
+    const result = await getVaultAndTransactions(address);
+    return result;
+  });
 
-  if (!vault && !isLoading && !isIdle) {
+  useEffect(() => {
+    if (vault) getOnchainVaultTransactions(vault).then(setVaultTxList);
```

Comment:
you could wrap this call in useQuery, and then you would get caching and be able to avoid `useEffect`, `useState`, & `then`. same applies to `VaultRow`. this will make clicking through from VaultRow to the full list of transactions immediate too, so long as you use the same cache key